components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -3342,6 +3342,21 @@ public APIStateChangeResponse changeLifeCycleStatus(String orgId, ApiTypeWrapper
             boolean isApiProduct = apiTypeWrapper.isAPIProduct();
             String workflowType;
 
+            if (!apiTypeWrapper.isAPIProduct()){
+                // validate custom API properties
+                if (StringUtils.equals(action, APIConstants.LC_PUBLISH_LC_STATE)) {
+                    org.json.simple.JSONArray customProperties = APIUtil.getCustomProperties(this.tenantDomain);
+                    List<String> errorProperties = APIUtil.validateMandatoryProperties(customProperties,
+                            apiTypeWrapper.getApi().getAdditionalProperties());
+
+                    if (!errorProperties.isEmpty()) {
+                        String errorString = " : " + String.join(", ", errorProperties);
+                        throw new APIManagementException(errorString, ExceptionCodes.from(ExceptionCodes
+                                .ERROR_WHILE_UPDATING_MANDATORY_PROPERTIES));
```

Comment:
Here we are not updating mandatory properties right? we are only doing a validation. Shall we introduce an exception code if one is not present already to reflect the same?